### ISSUE TITLE
Support for Planned Maintenance for `azurerm_kubernetes_cluster` (Preview)

### ISSUE THREAD

**Issue by aristosvo** [hashicorp/terraform-provider-azurerm#12107]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

An AKS cluster has regular maintenance performed on it automatically. By default, this work can happen at any time. Planned Maintenance allows you to schedule weekly maintenance windows that will update your control plane as well as your kube-system Pods on a VMSS instance and minimize workload impact. Once scheduled, all your maintenance will occur during the window you selected. You can schedule one or more weekly windows on your cluster by specifying a day or time range on a specific day. 

### New or Affected Resource(s)

* azurerm_kubernetes_cluster
* * azurerm_kubernetes_cluster_maintenance_configuration

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "example" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_D2_v2"
  }

  identity {
    type = "SystemAssigned"
  }
}

resource "azurerm_kubernetes_maintenance_configuration" "example" {
  name                = "example"
  cluster_name        = azurerm_kubernetes_cluster.example.name
  resource_group_name = azurerm_resource_group.example.name

  maintenance_allowed = {
    day = "Friday"
    hour_slots = [ 4 ]
  }

  maintenance_not_allowed_window = {
    start = "2021-05-26T03:00:00Z"
    end   = "2021-05-30T12:00:00Z"
  }
}
```

### References

- https://docs.microsoft.com/en-us/azure/aks/planned-maintenance

### Remarks
I'm more than happy to implement it myself

**Comment by njuCZ** [comment]
@aristosvo sorry for my duplicate work. I just noticed that you would like to implement it yourself. If you are in the process, I will close my PR.

**Comment by aristosvo** [comment]
No, please, continue!

**Comment by njuCZ** [comment]
thank you!

**Comment by heoelri** [comment]
I think this was released in [v2.71.0](https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v2.71.0) (PR https://github.com/hashicorp/terraform-provider-azurerm/pull/12762) and can be closed.
